crates/starknet/src/unit.rs

```diff
@@ -132,3 +132,26 @@ impl Unit {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_convert_u256_into_amount() {
```

Comment:
extract this logic as a function, and use it on many different values:
0, U256::MAX, u128::MAX, u128::MAX +1, 1e18, 1e17, 1e16